ldcpy/util.py

```diff
@@ -125,7 +128,15 @@ def preprocess(ds, varnames):
     return ds[varnames]
 
 
-def compare_stats(full_ds, varname, set1, set2, time=0, lev=0, significant_digits=4):
+def compare_stats(
+    ds,
+    varname,
+    set1,
+    set2,
+    significant_digits=4,
+    include_ssim_metric=True,
+    **metrics_kwargs,
```

Comment:
I removed the subsetting along `time`, `lev` so as to make this function generic. So far, for this function to work properly, the user must subset the data themselves:

So, with this change, this line

```python
ldcpy.compare_stats(col_ds, "TS", "orig", "zfpA1.0", time=0)
```

becomes 


```python
ds = col_ds.isel(time=0)
ldcpy.compare_stats(ds, "TS", "orig", "zfpA1.0", aggregate_dims=['lat', 'lon'])
```

I also exposed the arguments for the `DatasetMetrics` class via the `metrics_kwargs`. This allows us to remove the special logic that was being used to handle `cam` and `pop` data. So, if the user is working with POP data, they would do something along these lines:


```python
ldcpy.compare_stats(ds, "O2", "orig", "zfpA1.0", aggregate_dims=['nlat', 'nlon'])
```

Reply:
@allibco Curious what you think about having the user subset the data themselves. We could improve the subset_data function to make it convenient for users who are not familiar with xarray, so instead of 

```python
ds = col_ds.isel(time=0)
```

they would use 

```python
ds = ldcpy.subset_data(col_ds, subset={'time':0})
```

This would let the user subset the data in multiple ways using a single function call, making it syntactically simpler. However, we would have to anticipate and build in each type of subset the users might want.

Reply:
I am ok with the user subsetting themselves with xarray as long as we have good examples. 

Can you aggregate across time also in compare stats? Just wondering...@pinarda let's have a meetup today - I'll schedule...

Reply:
I like the idea of improving the capability if subset data for more novice users and advanced folks can use xarray if preferred.

What do you think @andersy005 ?

Reply:
> Can you aggregate across time also in compare stats? Just wondering 

Yes. 

> ds = ldcpy.subset_data(col_ds, subset={'time':0})

Subsetting the data for the user may end being too complicated down the line... For example, today we are using value based indexing via `sel` on `lat` and  array index based index via `isel` on `time`

https://github.com/NCAR/ldcpy/blob/ce17a8490e0e734a663532561870d50d5847c766/ldcpy/util.py#L393-L395

https://github.com/NCAR/ldcpy/blob/ce17a8490e0e734a663532561870d50d5847c766/ldcpy/util.py#L413-L416

So, if the user sees this `ds = ldcpy.subset_data(col_ds, subset={'time':0, 'lat': 20})`, I don't think it will be obvious to them that 

- `{'time':0}` corresponds to the first index on the time axis
- `{'lat': 20}` corresponds to `latitude=20` and not the 21st index on the latitude axis

Reply:
Hmmm...I see what you mean...it could be complicated later.